### PR TITLE
fix(gaussdb/cassandra): fix reducing nodes issue for prepaid resource

### LINKED ISSUE #2069: [Cassandra] fix reducing node issue for prepaid instance
Need to add order status check and cassandra insance actions check for prepaid resources otherwise it will cause issue when reducing nodes.

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #2069

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/ TESTARGS='-run=TestAccGeminiDBInstance'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/ -v -run=TestAccGeminiDBInstance -timeout 360m -parallel 4
=== RUN   TestAccGeminiDBInstanceDataSource_basic
=== PAUSE TestAccGeminiDBInstanceDataSource_basic
=== RUN   TestAccGeminiDBInstancesDataSource_basic
=== PAUSE TestAccGeminiDBInstancesDataSource_basic
=== RUN   TestAccGeminiDBInstance_basic
=== PAUSE TestAccGeminiDBInstance_basic
=== CONT  TestAccGeminiDBInstanceDataSource_basic
=== CONT  TestAccGeminiDBInstance_basic
=== CONT  TestAccGeminiDBInstancesDataSource_basic
--- PASS: TestAccGeminiDBInstancesDataSource_basic (1157.60s)
--- PASS: TestAccGeminiDBInstance_basic (1168.58s)
--- PASS: TestAccGeminiDBInstanceDataSource_basic (1330.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1330.425s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go [hunk 1]
```diff
@@ -806,19 +806,38 @@ func resourceGeminiDBInstanceV3Update(d *schema.ResourceData, meta interface{})
...
-				if result.Err != nil {
-					return fmtp.Errorf("Error shrinking huaweicloud_gaussdb_cassandra_instance %s node size: %s", d.Id(), result.Err)
+			for i := 0; i < loopSize; i++ {
+				if lastNum > 0 && (i == loopSize-1) {
+					reduceNum = lastNum
+				}
+				reduceNodeOpts := instances.ReduceNodeOpts{
+					Num: reduceNum,
+				}
+				logp.Printf("[DEBUG] Reduce Node Options: %+v", reduceNodeOpts)
```
Comment (Lance52259) [inline huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go]: `log.Printf` just find.

File: huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go [hunk 2]
```diff
@@ -806,19 +806,38 @@ func resourceGeminiDBInstanceV3Update(d *schema.ResourceData, meta interface{})
...
+
+				n, err := instances.ReduceNode(client, d.Id(), reduceNodeOpts).Extract()
+				if err != nil {
+					return fmtp.Errorf("Error shrinking huaweicloud_gaussdb_cassandra_instance %s node size: %s", d.Id(), err)
```
Comment (Lance52259) [inline huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go]: we should remove `huaweicloud` word from our error msg and start with a lowercase letter.

File: huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go (hunk 1 above)
Comment (niuzhenguo) [reply huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go]: done

File: huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go (hunk 2 above)
Comment (niuzhenguo) [reply huaweicloud/resource_huaweicloud_gaussdb_cassandra_instance.go]: done

**Pull request merged**